cli/lib/tasks/verify.js

```diff
@@ -10,90 +9,75 @@ const logSymbols = require('log-symbols')
 
 
 const { throwFormErrorText, errors } = require('../errors')
-const fs = require('../fs')
 const util = require('../util')
 const logger = require('../logger')
 const xvfb = require('../exec/xvfb')
 const state = require('./state')
 
-const verificationError = (message) => {
-  return _.extend(new Error(''), { name: '', message, isVerificationError: true })
-}
-
-const xvfbError = (message) => {
-  return _.extend(new Error(''), { name: '', message, isXvfbError: true })
-}
-
-const isMissingExecutable = (binaryDir) => {
+const checkExecutable = (binaryDir) => {
   const executable = state.getPathToExecutable(binaryDir)
   debug('checking if executable exists', executable)
-  return fs.pathExistsAsync(executable)
-  .then((exists) => {
-    if (!exists) {
-      return throwFormErrorText(errors.missingApp(binaryDir))(stripIndent`
+  return util.isExecutableAsync(executable)
+  .then((isExecutable) => {
+    debug('Binary is executable? :', isExecutable)
+    if (!isExecutable) {
+      return throwFormErrorText(errors.missingApp(executable))(stripIndent`
+          Failed to gain executable permissions on the Cypress binary at ${executable}.
+
+          Check that you have both downloaded and executed Cypress as the same User with the same permissions.
```

Comment:
This is probably the most cryptic error message. I'm not sure how to improve this exactly. 

Maybe it's the fact that you say "check" that you've downloaded and executed as same user. What does that mean exactly? How do you check that? 

Like, what's the actual action I have to take so that this executable permissions error does not show up next time.

Reply:
Alright, maybe I'll just remove the `Check that you have...`

Reply:
I think just telling the user we couldn't get executable permissions is enought

Reply:
```
Cypress cannot run because the binary does not have executable permissions: ${executable}
 
This is unfortunately a problem with your system.

We've seen this happen because:

- You've installed node as 'root' or with 'sudo'
- You've installed the cypress npm package as 'root' or with 'sudo'

Please check that you have the appropriate user permissions.
```